src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ReadyToRunSignature.cs

```diff
@@ -364,15 +366,25 @@ private string EmitString(StringHandle handle)
         }
     }
 
+    public interface IR2RSignatureTypeProvider<TType, TMethod, TGenericContext> : ISignatureTypeProvider<TType, TGenericContext>
+    {
+        public TType GetCanonType();
```

Comment:
It looks like we don't need to declare the method as public here.
This won't compile if we set the language version to be 7.3

Reply:
:) I've been accidentally adding public to interface methods since C# 1.0. As of the latest compilers it finally just lets me do it. I find it satisfying. (I will admit, when I was in the language design meeting where we added the feature I was kinda happy, even though it means that I will accidentally violate the coding standard somewhat often.)

However, as you say, it certainly isn't needed. I'll remove them.